File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.java
class Solution {
    public int solution(int n, int w, int num) {
        int depth = 0;
        int numRow = (num % w == 0)
            ? (num / w - 1)
            : (num / w);
        int originDirection = numRow % 2;
        int base = ((num + w - 1) / w) * w;
        int step = (base - num) * 2 + 1;

        while (num <= n) {
            depth++;

            if (numRow % 2 == originDirection) {
                num += step;
            } else {
                num += (w * 2) - step;
            }

            numRow++;
        }

        return depth;
    }
}

File: 프로그래머스/1/389478. 택배 상자 꺼내기/택배 상자 꺼내기.kt
class Solution {
    fun solution(n: Int, w: Int, num: Int): Int {
        var depth = 0
        var currentNum = num
        var numRow = if (num % w == 0) (num / w - 1) else (num / w)
        val originDirection = numRow % 2
        val base = ((num + w - 1) / w) * w
        val step = (base - num) * 2 + 1

        while (currentNum <= n) {
            depth++

            currentNum += if (numRow % 2 == originDirection) {
                step
            } else {
                (w * 2) - step
            }

            numRow++
        }

        return depth
    }
}